src/Appwrite/Auth/Validator/PasswordDictionary.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace Appwrite\Auth\Validator;
+
+use Utopia\Database\Document;
+
+/**
+ * Password.
+ *
+ * Validates user password string
+ */
+class PasswordDictionary extends Password
+{
+    protected array $dictionary;
+    protected Document $project;
+
+    public function __construct(array $dictionary, Document $project)
```

Comment:
We should always strive to simplify our signatures and use primitives when can. In this example, we can use an array of strings.

Reply:
I thought array_key_exists was more performant than in_array, so I used array_flip to get strings into keys. 10k strings are a lot.

Reply:
I was referring to the 2nd argument, the project document. Can we only get the relevant array from it?

Reply:
updated